homeassistant/components/sunweg/config_flow.py

```diff
@@ -39,11 +43,17 @@ async def async_step_user(self, user_input=None) -> FlowResult:
         if not user_input:
             return self._async_show_user_form()
 
-        # Initialise the library with the username & password
-        self.api = APIHelper(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
-        login_response = await self.hass.async_add_executor_job(self.api.authenticate)
-
-        if not login_response:
+        if self.api:
+            # Set username and password
+            self.api.username = user_input[CONF_USERNAME]
+            self.api.password = user_input[CONF_PASSWORD]
+        else:
+            # Initialise the library with the username & password
+            self.api = APIHelper(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
+        try:
+            await self.hass.async_add_executor_job(self.api.authenticate)
+        except SunWegApiError:
+            _LOGGER.error("Authentication error")
```

Comment:
Are these changes now still needed if reauth is excluded from this PR?

Reply:
This is not related. It just avoid creating another  APIHelper instance.  I can move it to another PR

Reply:
I'd prefer it in a separate PR

Reply:
Like if we can scope this PR to the dependency bump + all changes that flow out of that, that would be better

Reply:
The new version adds the setter for username and password. So, this change is related to the dependency bump, although not necessary.

Reply:
Can we scope this PR to the absolute necessary?

Maybe I should be a bit clearer next time, but this PR did too much at the same time. Let scope this PR to bumping the dependency and then do other stuff in other PRs

Reply:
done